nym-vpn-core/nym-vpn-lib/src/platform/mod.rs

```diff
@@ -130,6 +145,7 @@ pub struct VPNConfig {
     #[cfg(target_os = "ios")]
     pub tun_provider: Arc<dyn crate::OSTunProvider>,
     pub credential_data_path: Option<PathBuf>,
+    pub tun_status_listener: Option<Arc<dyn TunnelStatusListener>>,
```

Comment:
If I understand correctly, this is `Option` type here just so that we can have it implemented on both kotlin and swift, so to not put pressure on syncing the two sides. I think once we have the implementation stabilised on both platforms, we should probably change this to just `Arc<dyn TunnelStatusListener>` and set the value inside `LISTENER` to `Some(config.tun_status_listener)` in the `runVPN` function (a change either @zaneschepke  or @rokas-ambrazevicius  could do, whichever does the last implementation). Or is there a front-end reason for this being optional?

Reply:
We get tunnel updates from the network extension, so we don't need this.

Reply:
I made this optional so Rokas can implement if he wants, but he does not have to because he does not need it right now. 

I made it cross platform capable so he could potentially implement this in the future if we add some new features that would require a callback (like live tunnel statistics, etc). We could then just extend this same listener with another callback function to serve these additional use cases for both platforms.